rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala

```diff
@@ -93,4 +91,42 @@ object Location {
 
       case LTLimbo => StoreFail
     }
+
+  def fetch(loc: Location, k: Ctxt, globalEnv: TblObject): Ob =
+    loc.genericType match {
+      case LTCtxtRegister(reg) =>
+        if (reg < k.reg.size) {
+          k.reg(reg)
+        } else {
+          Ob.INVALID
+        }
+
+      case LTArgRegister(argReg) =>
+        if (argReg < k.argvec.elem.size) {
+          k.argvec.elem(argReg)
+        } else {
+          Ob.INVALID
+        }
+
+      case LTLexVariable(ind, level, offset) =>
+        k.env.getLex(ind, level, offset)
+
+      case LTAddrVariable(ind, level, offset) =>
+        k.env.getAddr(ind, level, offset)
+
+      case LTGlobalVariable(offset) =>
+        if (offset < globalEnv.numberOfSlots()) {
+          globalEnv.slot(offset)
+        } else {
+          Ob.INVALID
+        }
+
+      case LTBitField(ind, level, offset, spanSize) =>
+        k.env.getField(ind, level, offset, spanSize)
```

Comment:
We don't need the `sign` field?

Reply:
Hrm.  There's no sign field when we store it, so I need to dig into what's going on here.